security-features/privilege-restriction/index.rst

```diff
@@ -53,6 +53,46 @@ Regression tests: `test-kernel-security.py <https://git.launchpad.net/qa-regress
 
-Regression tests: `test-kernel-security.py <https://git.launchpad.net/qa-regression-testing/tree/scripts/test-kernel-security.py>`_.
+Access rules are expressed as::
+
+    <subject-label> <object-label> <access>
+
+where access describes the access permitted using the traditional Linux read (R or r) access, write (W or w) access, execute (X or x) access, or append (A or a) access. A dash (-) can be used as a place holder or to express that no access be permitted if used by itself.
+
+SMACK uses several special system labels:
+
+- ``_`` pronounced "floor"
+- ``*`` pronounced "star"
+- ``^`` pronounced "hat"
+
+There are a limited number of pre-defined rules:
+
++---------------+---------------+--------+
+| Subject Label | Object Label  | Access |
++===============+===============+========+
+| \*            | any           | \-     |
++---------------+---------------+--------+
+| any           | \*            | rwxa   |
++---------------+---------------+--------+
+| ordinary      | ordinary      | rwxa   |
++---------------+---------------+--------+
+| any           | \_            | rx     |
++---------------+---------------+--------+
+| ^             | any           | rx     |
++---------------+---------------+--------+
+
+The third rule uses "ordinary" to refer to any label except ``*`` and describes the case where the subject label and the object label are the same.
+
+Compared to `AppArmor <https://documentation.ubuntu.com/server/how-to/security/apparmor/index.html>`_, Ubuntu's default MAC system, SMACK and AppArmor serve different architectural philosophies. AppArmor uses path-based mandatory access controls focused on confining specific applications through profiles that restrict file access, network usage and capabilities. SMACK provides system-wide label-based access control that is more suitable for creating isolated security domains and enforcing consistent policies across all system components. AppArmor excels at application-specific confinement and is easier to deploy incrementally, while SMACK is better suited for environments requiring comprehensive labeling schemes, such as multi-tenant systems or devices where all processes and data need clear security classifications.
+
+SMACK support is available in Ubuntu kernels but is not enabled by default, as AppArmor serves as Ubuntu's primary LSM. To enable SMACK you need to add ``security=smack`` to the kernel line in ``/boot/grub/menu.lst``
```

Comment:
Just out of curiosity on my end, was SMACK support always available alongside Ubuntu? Or was it introduced with a certain kernel revision that corresponds to a named release? If so, it would be great to list when it was added here (for example, SMACK support is available in Ubuntu kernels starting with Ubuntu 18.04 LTS (Bionic Beaver)).